database/import.go

```diff
@@ -102,44 +104,79 @@ func dropHeadersFile(tmpHeadersFile *os.File, tmpHeadersFilePath string, log *ze
 	}
 }
 
-func parseRecord(record []string, rowIndex int32, previousBlockHash string) dto.DbBlockHeader {
-	hash := parseChainHash(record[0])
-	version := parseInt(record[1])
-	merkleroot := parseChainHash(record[2])
-	nonce := parseInt(record[3])
-	bits := parseInt(record[4])
-	chainWork := parseBigInt(record[5])
-	timestamp := parseInt64(record[6])
-	cumulatedWork := parseBigInt(record[7])
+func PrepareRecord(record []string, previousBlockHash string, bh service.BlockHasher, cumulatedChainWork string, rowIndex int) (dto.DbBlockHeader, error) {
+	parsedRow, err := parseRecordToBlockHeadersSource(record, previousBlockHash)
+	if err != nil {
+		return dto.DbBlockHeader{}, err
+	}
+	cumulatedChainWorkBigInt := parseBigInt(cumulatedChainWork)
+	rowIndexInt32 := int32(rowIndex)
```

Comment:
I would put these 2 lines into `calculateFields` method, will be a bit cleaner :)

Reply:
moved to calculateFields